src/openrct2/interface/window.c

```diff
@@ -403,6 +403,16 @@ void window_set_window_limit(sint32 value)
  */
 rct_window *window_create(sint32 x, sint32 y, sint32 width, sint32 height, rct_window_event_list *event_handlers, rct_windowclass cls, uint16 flags)
 {
+	//Number must always be kept within valid range.
+	if (gConfigGeneral.window_limit < WINDOW_LIMIT_MIN) {
+		gConfigGeneral.window_limit = WINDOW_LIMIT_MIN;
+		config_save_default();
+	}
+	if (gConfigGeneral.window_limit > WINDOW_LIMIT_MAX) {
+		gConfigGeneral.window_limit = WINDOW_LIMIT_MAX;
+		config_save_default();
+	}
```

Comment:
@IntelOrca @Gymnasiast You guys make a point.. a question though, I havent explored just yet, but does the config file ever get loaded again after the opening initialisation? window_set_window_limit handles the clamping when it's set in game, but not outside the game.